### PR TITLE
CI: Support uploading tarball to release

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
fixes https://github.com/lfortran/lfortran/issues/2590

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
It worked for me uploading the tarball here https://github.com/Shaikh-Ubaid/lfortran/releases/tag/v0.16.0.

I am hoping it might work here as well.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -828,6 +828,14 @@ jobs:
         env:
           SSH_DEPLOY_KEY: ${{ secrets.SSH_DEPLOY_KEY }}
 
+      - name: Upload Tarball to Release
+        if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+        shell: bash -e -l {0}
+        run: |
+          ci/upload_tarball_to_release.sh
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (Shaikh-Ubaid) [inline .github/workflows/CI.yml:837]: I am unsure if the above `GITHUB_TOKEN` would need any permission access to upload/edit releases. We might need to try and see with the default access.

**Comment by certik** [comment]
How can we try that this works without having to merge and release?

Or do you think it will work? I can create the token.

**Comment by certik** [comment]
Ok, let's merge it and see what happens. Thanks!

File: .github/workflows/CI.yml (hunk 1 above)
Comment (Shaikh-Ubaid) [reply .github/workflows/CI.yml:837]: The above is yet to be tested. It should hopefully work. If it fails, we need to either give access to the `GITHUB_TOKEN` or use a personal access token.

**Pull request merged**